### PR TITLE
Fix CRY generator. Select only one intersection with the world

### PR DESCRIPTION
**Pull request by oksuzian**
Before, we could have events with two intersections, which was unphysical

### CODE REVIEW

File: EventGenerator/src/CosmicCRY.cc [hunk 1]
```diff
@@ -211,29 +211,22 @@ void CosmicCRY::generate(GenParticleCollection& genParts) {
...
+          if(_worldIntersections.size()==0)
+            throw cet::exception("Rethrow") << "No intersection with the world. Can't happen. Abort.\n";
+          //Find the iterator that has the highest intersaction with the world
+          //Very rare cases of bottom intersections are below the projection and unphysical
+          auto highest_worldIntersection = std::max_element(_worldIntersections.begin(), _worldIntersections.end(),
+                                        [](const CLHEP::Hep3Vector& a, const CLHEP::Hep3Vector& b) {
+                                          return a.y() < b.y();
+                                        });
+          //Dereference the iterator to find the projected Hep3Vector position from the world
+          Hep3Vector projectedPos = *highest_worldIntersection;
```
Comment (kutschke) [inline EventGenerator/src/CosmicCRY.cc:226]: I think this should be 

`Hep3Vector const&`

**Review by kutschke — CHANGES_REQUESTED** [review]
Looks OK to me.  Please make the one small change.

File: EventGenerator/src/CosmicCRY.cc (hunk 1 above)
Comment (oksuzian) [reply EventGenerator/src/CosmicCRY.cc]: Thanks for looking into this, @kutschke .
I've implemented your suggestion.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Pull request merged**